grpc/server/src/main/java/io/helidon/grpc/server/ProtoReflectionService.java

```diff
@@ -110,7 +112,7 @@ public static BindableService newInstance() {
      * mutable services or a change in the service names.
      */
     private ServerReflectionIndex getRefreshedIndex() {
-        synchronized (lock) {
+        return accessIndex(() -> {
```

Comment:
Perhaps prefer:
```
// change indexAccess to a simple Lock
indexAccess.lock();
try {
  Server server = ...
} finally {
  indexAccess.unlock();
}
```
…which is the idiom I was naïvely expecting to see here.

Reply:
That's the construction used in the updated separate method.